proposals/0000-split-jest-from-react-native-core.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Split Jest from React Native Core
+author:
+- Phil Pluckthun
+- Cedric van Putten
+date: 2025-08-17
+---
+
+# RFC0000: Split Jest from React Native Core
+
+## Summary
+
+This RFC proposes the removal of `react-native/jest-preset` from the core React Native package, and moving it to `@react-native/jest-preset` instead, making Jest an optional testing framework that developers can choose to integrate only if desired. This change would acknowledge the ongoing changes in test runner preferences and ease maintenance.
+
+## Basic example
+
+Currently, `react-native` itself contains a Jest preset and after setting up a project is available at `react-native/jest-preset`.
+
+```json
+{
+  "dependencies": {
+    "react-native": "0.81.x"
+  },
+  "devDependencies": {
+    "jest": "*"
+  }
+}
+```
+
+The proposed change is to split this out into a separate, `@react-native/jest-preset` package, thus;
+
+```json
+{
+  "dependencies": {
+    "react-native": "0.81.x"
+  },
+  "devDependencies": {
+    "@react-native/jest-preset": "0.81.x",
+    "jest": "*"
+  }
+}
+```
+
+## Motivation
+
+There are several key motivations for this change:
+
+- **Maintenance Status:** While Jest 30 has now been released after a long break of major maintenance updates, [it includes a regression for `transformIgnorePatterns`](https://github.com/jestjs/jest/issues/15781), demonstrating that React Native isn't Jest's main focus, and that continuously maintaining a Jest preset in React Native that is part of the core package poses challenges.
+- **Ecosystem Evolution:** Outside of Meta, many teams and projects have migrated to alternative testing solutions, especially for unit testing, such as Vitest. These days alternative test runners exist that are better set up to support React Native in the future, that the ecosystem signals they're starting to prefer. By bundling the Jest preset by default, we create an artificial constraint and recommendation that doesn't reflect current practices.
```

Comment:
I agree and love this, but personally every time i have looked for setting up something like Vitest on React Native, the info out there is pretty minimal. Maybe also as part of this work we could also add some links to https://reactnative.dev/docs/testing-overview#writing-tests for Jest and alternatives?